src/bosh-director/lib/bosh/director/api/tasks_config_manager.rb

```diff
@@ -0,0 +1,165 @@
+module Bosh::Director
+  module Api
+    class TasksConfigManager
+      include ValidationHelper
+      WORKER_DEFAULT_LIMIT = 100
+
+      def add_to_groups(delayed_job)
+        Transactor.new.retryable_transaction(Bosh::Director::Config.db) do
+          configs = tasks_configs
+          unless configs.empty?
+            payload_object = delayed_job.send(:payload_object)
+            task_id = payload_object.task_id
+            tasks_configs.each do |tasks_config|
+              begin
+                delayed_group = tasks_config[:delayed_group]
+                delayed_group.add_delayed_job(delayed_job) if tasks_config[:tasks_config].applies?(task_id)
+              rescue Sequel::ValidationFailed, Sequel::DatabaseError => e
+                error_message = e.message.downcase
+                if error_message.include?('unique') || error_message.include?('duplicate')
+                  next
+                else
+                  raise e
+                end
+              end
+            end
+          end
+        end
+      end
+
+      def rebuild_groups
+        cleanup
+
+        raw_tasks_configs.each do |raw_tasks_config|
+          Bosh::Director::Models::DelayedJobGroup.create(
+            config_content: raw_tasks_config.tasks_config_text,
+            limit: raw_tasks_config.is_rate_limit? ? raw_tasks_config.rate_limit : WORKER_DEFAULT_LIMIT
+          )
+        end
+
+        delayed_jobs = Delayed::Backend::Sequel::Job.where(::Sequel.expr(:failed_at => nil)).all
+        delayed_jobs.each do |delayed_job|
+          add_to_groups(delayed_job)
+        end
+      end
+
+      private
+
+      def cleanup
+        Models::DelayedJobGroup.dataset.delete
+      end
+
+      def raw_tasks_configs
+        tasks_configs = ConfigManager.new.find(type: 'tasks', name: nil, latest: true)
+        parsed_configs = []
+        if !tasks_configs.nil? && !tasks_configs.empty?
+          content_hash = YAML.load(tasks_configs[0].content)
+          tasks_config_hashes = safe_property(content_hash, 'tasks', :class => Array, :default => [])
```

Comment:
let's rename `tasks` key to `rules`. i've updated bosh-notes as well: https://github.com/cloudfoundry/bosh-notes/blob/master/proposals/tasks-config.md.